### PR TITLE
docs: show benchmark results for storage capacity tracking

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

This was done in preparation for promoting storage capacity tracking to GA. The
csi-driver-hostpath repo seems like a suitable place to hold this text because
it showcases how to use the distributed deployment example.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: docs/storage-capacity-tracking.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+```
+I0307 08:36:55.412124    6877 simple_test_executor.go:145] Step "[step: 03] Waiting for deployments to be running" started
+W0307 08:45:52.536411    6877 reflector.go:436] *v1.PodStore: namespace(test-pu6g95-10), labelSelector(name=deployment-652): watch of *v1.Pod ended with: very short watch: *
+v1.PodStore: namespace(test-pu6g95-10), labelSelector(name=deployment-652): Unexpected watch close - watch lasted less than a second and no items received
+...
+```
+
+There were other intermittent problems accessing the apiserver. Tweaking the
+way how external-provisioner creates CSIStorageCapacity objects solved this
+problem and the same test passed all three times that it was run:
```
Comment (jsafrane) [inline docs/storage-capacity-tracking.md:283]: What tweaks did you use?

Comment (pohly) [reply docs/storage-capacity-tracking.md]: That was meant to be a link to https://github.com/kubernetes-csi/external-provisioner/pull/711. I just used the defaults from that updated external-provisioner - text updated.

**Comment by pohly** [comment]
/hold

Let me add results for 100 nodes with pods that have no volumes...

**Comment by pohly** [comment]
/hold cancel

Baseline for 100 nodes was added.

The overhead for pods with volumes was actually slightly lower than it was for 10 nodes.

**Pull request merged**